### DISCUSSION TITLE
Support for ImageMosaic for raster time-series data

### DISCUSSION THREAD

**Discussion by ammaciel** [eblondel/geosapi#71] [Q&A] [answered]
The package developed is very interesting, and I am studying more about it. I'm currently working with the Geoserver ImageMosaic plugin and would like to automate this task in R.

My data are composed of a set of raster data with different time periods. In this case, I need to configure a mosaic directory (https://docs.geoserver.org/main/en/user/tutorials/imagemosaic_timeseries/imagemosaic_timeseries.html).

I created a ImageMosaic coveragestore without problem (GSImageMosaicCoverageStore$new), but my issue is in relation to add a directory with a set of raster with different timestamps. For instance:

$ls path/max_temperatures/*.tif

max_temperatures/temperature_20221215.tif
max_temperatures/temperature_20221216.tif
max_temperatures/temperature_20221217.tif
max_temperatures/temperature_20221218.tif
max_temperatures/temperature_20221219.tif

I tried to use the Method uploadImageMosaic() with a directory, and it returns the error _cannot open file '<path_dir>': it is a directory_. Furthermore, I had checked package documentation and other files and not discover a solution that can work with my data. Appreciate if you can help me with this issue.

**Comment by kikislater** [comment]
Geonode do it with geoserver but index its tiles with dbms (postgres/postgis) like you mention with the geoserver documentatiin link.
So, getting inspiration from IMO with ImageMosaic and API could save time for future intégration in geosapi ...

**Comment by eblondel** [comment, marked as answer]
Dear @ammaciel first of all my apologies for not having seen your issue
To upload an image mosaic set of file, you use the right method `uploadImageMosaic`, but you should put your files in a zip archive. Did you try it?
